test/mock/common/context/context.go

```diff
@@ -0,0 +1,84 @@
+// Code generated by MockGen. DO NOT EDIT.
+// Source: context.go
+
+// Package mock_mock_context is a generated GoMock package.
+package mock_context
```

Comment:
I'm curious how you created this? By referencing code in vendor dir?

Reply:
I tried that but it proven difficult. I ended up putting the context interface into a new file, generated the mock and removed the file.

I wanted to avoid mocking code we didn't write by creating a context with the value we needed. I hit a dead end when I noticed that the key for storing/retrieving that value was an struct from the peer package that it is never exposed. https://github.com/grpc/grpc-go/blob/master/peer/peer.go#L49

Reply:
Hmmm that's a total bummer. I feel this will bite us at some point, but unfortunately don't have any better suggestions.

Did you check in the interface somewhere? If not, we should, and comment it heavily. Makes me wonder if we should use that interface too in the method you're passing ctx into

Reply:
Yep, I didn't like it either that's why I tried the other approach first. 
The method already uses the interface as a parameter type so no need to change that. 
I removed the interface but I can add it once more, it is a simple one, 4 methods.